File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
  let left = 0;
  let right = height.length - 1;
  let max = 0;
  
  while (left < right) {
    const area = (right - left) * Math.min(height[left], height[right]);
    
    max = Math.max(max, area);
    
    if (height[left] > height[right]) {
      right--;
    } else {
      left++;
    }
  }
  
  return max;
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
  def maxArea(self, height: List) -> int:
    left = 0
    right = len(height) - 1
    maximum = 0

    while left < right:
      x_axis = right - left
      y_axis = min(height[left], height[right])
      area = x_axis * y_axis

      maximum = max(maximum, area)

      if height[left] < height[right]:
        left += 1
      else:
        right -= 1

    return maximum
